api/v1/applicationlayer_types.go

```diff
@@ -36,11 +36,17 @@ type ApplicationLayerSpec struct {
 	// L7LogCollectorDaemonSet configures the L7LogCollector DaemonSet.
 	// +optional
 	L7LogCollectorDaemonSet *L7LogCollectorDaemonSet `json:"l7LogCollectorDaemonSet,omitempty"`
+
+	// SidecarInjection controls whether or not sidecar injection is enabled for the cluster.
+	// When enabled, sidecar containers can be injected into pods to provide additional L7 functionality.
+	// +optional
+	SidecarInjection *SidecarStatusType `json:"sidecarInjection,omitempty"`
```

Comment:
It's not super clear to me how this field interacts with the other fields in here.

A few questions I have:
- Is it valid to have `SidecarInjection: Enabled` and all the other fields be Disabled? If so, what is the intended behavior of the system under that configuration?
- What is the "additional L7 functionality" this provides? My understanding was that sidecar was just a deployment technique for all of the existing L7 functionality (WAF, logs, ALP, etc).